content/source/docs/enterprise/sentinel/import/tfstate.html.md

```diff
@@ -10,103 +10,526 @@ description: |-
+of all of the modules within the Terraform state at plan-time.
 
-This returns all the resources in the module with the given type
-as a map from name to resource. For the documentation on the
-value, see `m.resources.TYPE.NAME`
+Note the distinction here - this means if a module is not present in state, it
+will not be present in `module_paths`.
 
-### m.resources.TYPE.NAME
+This data is represented as a list of a list of strings, with the inner list
+being the module address, split on the period (`.`).
 
-This returns the list of resources with the given type and name. The
-result is a list because the plan is aware that the count may be greater
-than zero.
+The root module is included in this list, represented as an empty inner list, as
+long as it is present in state.
 
-### m.data
+As an example, if the following module block was present within a Terraform
+configuration:
 
-This returns all the data sources in the module as a map. The map key is the
-type, the value is the same as `m.data.TYPE`.
+```hcl
+module "foo" {
+  # ...
+}
+```
 
-### m.data.TYPE
+The following policy would evaluate to `true`, _only_ if the diff had changes
+for that module:
 
-This returns all the data sources in the module with the given type as a map
-from name to data source. For the documentation on the value, see
-`m.data.TYPE.NAME`
+```python
+import "tfstate"
 
-### m.data.TYPE.NAME
+main = rule { tfstate.module_paths contains ["foo"] }
+```
 
-This returns a list of data sources and behaves exactly how a resource lookup
-behaves with the m.resources.TYPE.NAME syntax above. All fields that are
-accessible within resources are available within a data source, except for
-`tainted`, as data sources cannot be tainted.
+#### Iterating through modules
 
-### m.outputs.NAME
+Here is an example of a function to retrieve all resources of a particular type
+from all modules (in this case, the
+[`azurerm_virtual_machine`][ref-tf-azurerm-vm] resource). Note the use of `else
+[]` in case some modules don't have any resources; this is necessary to avoid
+the function returning undefined.
 
-Returns the output specified by NAME. This works in submodules as well.
+[ref-tf-azurerm-vm]: /docs/providers/azurerm/r/virtual_machine.html
 
-## Type: o
+Remember again that this will only locate modules (and hence resources) that are
+present in state.
 
-### o.value
+```python
+import "tfstate"
 
-The output value
+get_vms = func() {
+	vms = []
+	for tfstate.module_paths as path {
+		vms += values(tfstate.module(path).resources.azurerm_virtual_machine) else []
+	}
+	return vms
+}
+```
 
-### o.sensitive
+### Value: `terraform_version`
 
-True if the output is marked as sensitive
+* **Value Type:** String.
 
-### o.type
+The `terraform_version` value within the [root namespace](#namespace-root)
+represents the version of Terraform in use when the state was saved. This can be
+used to enforce a specific version of Terraform in a policy check.
```

Comment:
I think this might also be copy-pasta. If the version is in the state, it cannot be used as an enforcement check since that already happened - it might make more sense to say it can be used to check if there is a large jump in the old vs new versions?

Reply:
This isn't a copy/paste, but indeed it's confusing. This is exposed in both the plan and state (and indeed saved in both as well). Which one do you check? Definitely the plan would be the better one, to be honest as it would technically be more current - so maybe an info box here would be a good idea noting this, keeping in mind that we might have future scenarios where a plan won't be available (example: periodic policy checks).

To be honest, I don't even know if the version is a good idea to expose in imports - I'd personally think this might be better in a `tfe` import or something along those lines that can get run data from TFE itself, which would be the ultimate source of truth.